tensorflow/workspace2.bzl

```diff
@@ -155,6 +155,18 @@ def _tf_repositories():
         ],
     )
 
+    tf_http_archive(
+        name = "cudnn_frontend_archive",
+        build_file = "//third_party:cudnn_frontend.BUILD",
+        patch_file = "//third_party:cudnn_frontend_header_fix.patch",
+        sha256 = "498f908ced41bbf524af6b89dc4229d5cc89311bfaaed1e3794981e858629196",
+        strip_prefix = "cudnn-frontend-360d6e7164dfb7c802493fd1c0464f0d815b852a",
+        urls = [
+            "https://storage.googleapis.com/mirror.tensorflow.org/github.com/NVIDIA/cudnn-frontend/archive/360d6e7164dfb7c802493fd1c0464f0d815b852a.zip",
+            "https://github.com/NVIDIA/cudnn-frontend/archive/360d6e7164dfb7c802493fd1c0464f0d815b852a.zip"
+        ],
+    )
+
```

Comment:
How hard would it be to remove the dependency on this library and use the cuDNN C interface directly?

Reply:
Directly using the C interface (i.e. the CUDNN backend APIs) will introduce many many more lines of code, since we need to explicitly handle the creation of the newly introduced descriptors: ops, opGraphs, engines, enginePlans, etc. Also, for each of them, there are bunch of attributes that we need to set one by one by using backend API calls. Besides, we have to be very careful about when we need to reclaim those objects, meaning for each object, we also need to put it in a unique_ptr with a dedicated deleter.

Reply:
My understanding is that if we don't use this dependency, we have to invent it by ourselves. StreamExecutor has a lot of RAII wrappers for cuDNN structs. As long as the dependency is header-only, we can depend on it as-if we invent it in the repository (no ABI issues).